rmgpy/rmgobject.pyx

```diff
@@ -133,3 +134,52 @@ cpdef expand_to_dictionaries(obj):
             return expand_to_dictionaries(obj.as_dict())
         else:
             return obj
+
+cpdef recursive_make_object(obj, class_dictionary, make_final_object=True):
+    if isinstance(obj, dict):
+        if 'class' in obj:  # This is a dictionary of an object to be created
+            try:
+                class_to_make = class_dictionary[obj['class']]
+            except KeyError:
+                raise KeyError('Class {0} must be provided in the class_dictionary: {1}'.format(obj['class'],
+                                                                                                class_dictionary))
+
+            for key, val in obj.iteritems():
+                if key == 'class':
+                    continue
+                obj[key] = recursive_make_object(val, class_dictionary)
+
+            args = {key:obj[key] for key in obj if key != 'class'}
+            if make_final_object:
+                if hasattr(class_to_make, 'make_object'):
+                    created_obj = class_to_make.__new__(class_to_make)
+                    created_obj.make_object(args, class_dictionary)
+                else:
+                    created_obj = class_to_make(**args)
+                return created_obj
+            else:
+                return args
+        else:
+            for key, val in obj.iteritems():
+                obj[key] = recursive_make_object(val, class_dictionary)
+            return obj
+
+    elif isinstance(obj, list):
+        return [recursive_make_object(x, class_dictionary) for x in obj]
+
+    elif isinstance(obj, str):
+        # Check to see if the string needs to be converted to a float or int
+        if bool(re.search('\d', obj)):
```

Comment:
Maybe search for `^\d+$` (start to end only digits) instead of the try except

Reply:
I don't think this works for floats (try `bool(re.search('^\d+$', '12.45'))` in python 2). I'll keep looking for a better method though.

Reply:
will `^[\d.]+$` work?

Reply:
Closer, but it doesn't catch floats in scientific notation like `'-1.20e-05'` and negative numbers

Reply:
Idk if this is a good solution, but could you just try casting to float, then int if that fails, then leave as string if that fails?

Reply:
Is there an instance where something could be cast as `int` but not `float`? Also, would this convert all `int`s to `float`s?

Reply:
Oh yeah, that's true. Is there a benefit to the first search for digits? Could you skip directly to the check for a decimal point?